hub/core/storage/local.py

```diff
@@ -94,7 +96,9 @@ def __delitem__(self, path: str):
             KeyError: If an object is not found at the path.
             DirectoryAtPathException: If a directory is found at the path.
             Exception: Any other exception encountered while trying to fetch the object.
+            ReadOnlyProviderError: If the provider is in read-only mode.
```

Comment:
Let's rename this. Most end-users will have no idea about what a provider is. We should instead just say that "the dataset is open in read mode"

Reply:
Good call

Reply:
@AbhinavTuli renamed to ReadOnlyError. Note that this docstring is for the _local storage provider_, so it makes sense to say "provider" here. The actual error contents don't refer to providers so this should be hidden.